pkgs/build-support/cc-wrapper/ld-wrapper.sh

```diff
@@ -121,7 +121,7 @@ if [ "$NIX_@infixSalt@_DONT_SET_RPATH" != 1 ]; then
     # It's important to add the rpath in the order of -L..., so
     # the link time chosen objects will be those of runtime linking.
     declare -A rpaths
-    for dir in "${libDirs[@]}"; do
+    for dir in ${libDirs+"${libDirs[@]}"}; do
```

Comment:
It is ugly, but it really has helped me catch all sorts of bugs---and we've been changing stdenv and cc-wrapper more lately than we have in years. My takeaway then is the maintainability benefit is ongoing, but the ugliness is temporary, so it's a clear win.